### ISSUE TITLE
Nightly builds of deb packages?

### ISSUE THREAD

**Issue by xet7** [Secretchronicles/TSC#666]
@boospy 

Now TSC has new statusbar at bottom of level editor. Can you provide new 32bit and 64bit DEB-Packages for 18.04 ?

I'm currently building new Windows packages.

**Comment by boospy** [comment]
Will build new package.

**Comment by xet7** [comment]
Thanks!

**Comment by xet7** [comment]
@boospy 

Now most of level editor etc is translated. I added new Windows packages:
https://secretchronicles.org/en/news/2019/11/21/Updated-Windows-Packages/

Do you have time to provide new 32bit and 64bit DEB-Packages for 18.04 ?

**Comment by Quintus** [comment]
Maybe it's easier to think about automatic nightly builds on
alexandria? Doing it manually all the time costs boospy lots of time
probably.

-- 
Blog: https://mg.guelker.eu

**Comment by xet7** [comment]
Can someone profide info how to build 32bit and 64bit DEB packages? I could setup automatic builds to alexandria, using qemu.

**Comment by xet7** [comment]
Or alternatively, I can setup building packages with free https://packager.io/ service that builds packages automatically and can be used as deb/rpm repo.

**Comment by boospy** [comment]
> Maybe it's easier to think about automatic nightly builds on alexandria? Doing it manually all the time costs boospy lots of time probably.
> […](#)
> -- Blog: https://mg.guelker.eu

This would be very nice. I will describe how i build. I'am not a dev, so i know this is awkward. I write that down.

**Comment by Quintus** [comment]
@boospy We have recently released [a beta1 of TSC 2.1.0](https://lists.secretchronicles.org/hyperkitty/list/tsc-devel@lists.secretchronicles.org/thread/LDVC5HUMGCXKSRTC4VHBJWKGUKI7TCYH/) and it'd be nice if you could provide packages for this. As we're going to release one or more betas before the final 2.1.0 release, it'd be good if you subscribed to the linked tsc-devel mailinglist so you don't miss those and can provide packages (if you find the time of course). Once the packages are available, we'll announce the beta release on the TSC website.

**Comment by boospy** [comment]
Is no online. 

How i build?

1. pull the repo
2. build the new version with ./build-tsc.sh
Here my file do build:

```
#!/bin/bash
 
# Get latest git submodules
git submodule init
git submodule update
 
# Change to tsc directory
cd tsc
 
# Delete old build files
rm -rf build
mkdir build
cd build
 
# Delete old game directory at that is at repos/Secretchronicles/TSC/tsc/tsc
#rm -rf ~/tsc
 
# Build TSC
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr ..
make -j8
 
# Install TSC to $HOME/tsc
make install
```
After install the game is on the local disk. Now i move the files to my builddirectory. Both attached. 
Here are the files: 
https://darkdevil.osit.cc/nextcloud/s/FogYy7RQTykXHfS?path=%2Fshares%20for%20forums%2FTSC

So and last. I played to a level the i found something that i do not know what i should do here... strange. Here is the picture: https://darkdevil.osit.cc/nextcloud/s/FogYy7RQTykXHfS?path=%2Fshares%20for%20forums%2FTSC#/shares%20for%20forums/TSC/tsc_what.png

:1st_place_medal: 

:) :) :)

**Comment by Quintus** [comment]
> Is no[w] online.

Thank you very much!

@sydneyjd Could you please pull the packages and upload them to the website? Once done I will create a news post for the beta1.

> How i build?

@xet7 Can you take a look at this instructions and tell me if this is automatable in a way we can implement on alexandria with a shell script or similar? Note we already have an automatically updated copy of the TSC repository on alexandria in `/srv/git/TSC.git` where you can easily retrieve the current source code from in a fast way without extra network overhead.

> So and last. I played to a level the i found something that i do not know what i should do here... 

May I ask to link pictures in a way that does not require JavaScript? You can perfectly well include pictures into GitHub comments by use of the `![alt text](/url/to/image.png)` syntax. GitHub will inline the image.

**Comment by xet7** [comment]
@Quintus here is the image from that bug @boospy mentioned:

So and last. I played to a level the i found something that i do not know what i should do here... strange. Here is the picture: 

![tsc_what](https://user-images.githubusercontent.com/15545/71326056-5f74ba80-24fe-11ea-8ee7-bd37f5815e3f.png)

**Comment by xet7** [comment]
@boospy 

> After install the game is on the local disk. Now i move the files to my builddirectory. Both attached.

Please give details what commands you used to build deb package. I did not find that info yet.

**Comment by Quintus** [comment]
> Here is the picture:

Thanks! Which level is that? When you're in the level, press F8 to open the editor and you can read the level name in the bottom-left corner in cyan blue.

In any case, the switch should have activated something.

**Comment by xet7** [comment]
I figured out a way to build deb package, but I'll test does it work, before uploading.

**Comment by xet7** [comment]
I have uploaded these deb packages to https://ftp.secretchronicles.org/releases/
- amd64 for Ubuntu 18.04
- i386 for Ubuntu 18.04
- armhf for Rasbian raspi3/raspi4

I did build manually on each distro desktop. Here is info how to manually build for any .deb distro architecture:
https://ftp.secretchronicles.org/build-deb/

I have not yet figured out how to automate building .deb packages.

**Comment by sydneyjd** [comment]
I have tested that Xet7's amd64 Package does install and run properly on Ubuntu 18.04. Would you still like me to pull Boospy's Ubuntu package and upload it to the website?

**Comment by xet7** [comment]
@sydneyjd 

You don't need to upload anything.

I did already create exe and deb packages and their checksums, and I will create those packages also in the future. Sometime I will also automate nightly builds. We don't need to wait when @boospy has time.

**Comment by Quintus** [comment]
Thanks, xet7. @boospy I hope that this is line with your expectations. You could do some testing of the beta1 release and see if you find bugs. Or even better, try building new levels! TSC is in urgent need of new quality levels. Please submit any levels you create to the linekd mailing list.

**Comment by boospy** [comment]
Today I installed the TSC 2.1.0-rc2 DEB package on my KDE NEON (ubuntu 20.04). Unfortunately I could not start the game because a dependency was not installed.

`tsc: error while loading shared libraries: libboost_chrono.so.1.71.0: cannot open shared object file: No such file or directory
`
After installing the libboost-chrono1.71.0 package. It started fine.

**Comment by xet7** [comment]
@boospy 

Yes, 3 days ago I added that libboost-chrono dependency to Ubuntu 20.10 package https://github.com/Secretchronicles/main-website/commit/4dec10e13c15d287b5d6ac51902b0db99c680cc8 but it's still missing from other packages, I will update other packages in near future.

**Comment by xet7** [comment]
@boospy 

Why old rc2? Official release of 2.1.0 is at https://secretchronicles.org/en/download/

**Comment by boospy** [comment]
Sorry my fault. Wrong packages in my repo. :) :)

**Comment by xet7** [comment]
I checked all official .deb packages, and it seems only sid package had libboost-chrono missing. I'm currently uploading sid package.

**Comment by xet7** [comment]
Sid package is now uploaded and md5+sha256 hashes updated on website.

**Comment by xet7** [comment]
I'll think about nightly builds, or builds when there is new commit.
